### PR TITLE
Fix/reduce announce traffic

### PR DESCRIPTION
**Pull request by mbg033**
- reduced announce traffic (relayed to the peers with probability = 1/N)
- RTA messages transferred over handshaked P2P connections

### CODE REVIEW

**Comment by mbg033** [comment]
rebased

File: src/p2p/net_node.inl [hunk 1]
```diff
@@ -123,6 +124,26 @@ namespace nodetool
...
 
+    /*!
+     * helper to return non-empty subset of 'in' container, each element included with probability 'p'
+     */
+    template <typename T>
+    void select_subset_with_probability(double p, const T &in, T &out)
+    {
+        if (in.empty())
+            return;
+        auto gen = std::mt19937{std::random_device{}()};
```
Comment (bitkis) [inline src/p2p/net_node.inl]: supposed to be static/global

File: src/p2p/net_node.inl [hunk 2]
```diff
@@ -841,6 +862,33 @@ namespace nodetool
       return true;
   }
 
+  //-----------------------------------------------------------------------------------
+  template<class t_payload_net_handler>
+  bool node_server<t_payload_net_handler>::get_random_connections_with_probability(double p, std::list<boost::uuids::uuid> &out_connections) const
+  {
+    auto f = [&](double p, std::list<boost::uuids::uuid> &out_connections)
+    {
+      auto gen = std::mt19937{std::random_device{}()};
```
Comment (bitkis) [inline src/p2p/net_node.inl]: supposed to be static/global

Comment (mbg033) [reply src/p2p/net_node.inl]: this method has been deleted  (unused)

**Pull request merged**